### DISCUSSION TITLE
[Chapter14] Front 소스코드 리팩토링 질문

### DISCUSSION THREAD

**Discussion by snoopingh** [cowaine/is-my-code-that-strange#110] [Chapter 13-14 (모델링, 리팩터링)] [answered]
리팩토링 챕터를 보고나서 진행중인 프론트 프로젝트에서 개선점을 찾아 적용해보려고 합니다. (Vue, React)

Router 공통 처리 부분에서 로그인 시에 회원 정보를 가져와 전역으로 상태 관리를 하게 하고,
저장한 정보를 각 화면에서 회원 관련 API 호출에 사용하고 있습니다.

회원과 관련된 화면에서 사용중인 API 호출을 각각 메서드로 만들어 한 파일로 모아 놓고
각 화면 컴포넌트에서는 필요한 메서드만 사용하게 하려고 합니다.

상태 관리 모듈에서 회원 관리와 대한 책임을 가지게 하고, 호출만 별도의 파일로 분리하는 것이
응집도를 높일 수 있을 것 같다고 생각했는데 이것이 적절한 방법일런지요?

**Comment by corock** [comment, marked as answer]
코드가 없어서 정확한 상황은 파악이 안 되지만 분리하는 게 일반적으로 많이 사용하는 방법 아닐까요?

React 를 예로 들면 Redux 나 Recoil 같은 상태 관리 라이브러리로 상태는 보통 store 문자열이 포함된 디렉터리로,
그리고 호출을 담당하는 라우터는 router 또는 api 문자열이 포함된 디렉터리에 분리하는 것으로 알고 있습니다.
